src/sdam/state/mod.rs

```diff
@@ -254,33 +270,36 @@ impl Topology {
             // in 4.2+, we only clear connection pool if we've received a
             // "node is shutting down" error. Otherwise, we always clear the pool.
             if wire_version < 8 || error.is_shutting_down() {
-                server.clear_connection_pool();
+                server.pool.clear();
             }
+
+            self.common.message_manager.request_topology_check();
         }
     }
 
     /// Marks a server in the cluster as unknown due to the given `error`.
-    async fn mark_server_as_unknown(&self, error: Error, address: StreamAddress) {
-        let description = ServerDescription::new(address, Some(Err(error)));
-        self.update(description).await;
+    async fn mark_server_as_unknown(
+        &self,
+        error: Error,
+        server: &Server,
+        state_lock: RwLockWriteGuard<'_, TopologyState>,
+    ) -> bool {
+        let description = ServerDescription::new(server.address.clone(), Some(Err(error)));
+        self.update_and_notify(description, state_lock).await
     }
 
-    /// Updates the topology using the given `ServerDescription`. Monitors for new servers will
-    /// be started as necessary.
-    ///
-    /// Returns true if the topology changed as a result of the update and false otherwise.
-    pub(crate) async fn update(&self, server_description: ServerDescription) -> bool {
+    async fn update_and_notify(
```

Comment:
Can this method get a doc comment too? I think the SDAM machinery can be tricky to understand for someone not already used to it, so having all these methods documented helps make that easier.

Reply:
Yep done.